### ISSUE TITLE
:bug: Check if there are race conditions when user token is renewed

### ISSUE THREAD

**Issue by penguineer** [penguineer/Rabbarkable#3]
The user token is renewed every 6 hours, but due to the asynchronous nature of the implementation there might be requests with the old and renewed token in parallel.

Check if this is a problem or if there can indeed be multiple tokens at the same time.

**Comment by penguineer** [comment]
There seems to be no problem  with multiple user tokens (tested by accidentally having two).

There is another issue tho:
When the token is expired, it takes a while to retrieve a new one. In this time other (parallel) requests will also issue a user token. Some synchronization might be beneficial here.

**Comment by penguineer** [comment]
The problem #5 still exists, but multiple tokens do not seem to be a problem.
